dartagnan/src/main/java/com/dat3m/dartagnan/encoding/WmmEncoder.java

```diff
@@ -638,17 +638,17 @@ public Void visitSyncBarrier(Relation rel) {
         @Override
         public Void visitSyncFence(Relation syncFence) {
             boolean idl = !context.useSATEncoding;
-            List<Fence> allFenceSC = program.getThreadEvents(Fence.class).stream()
-                    .filter(e -> e.hasTag(Tag.PTX.SC))
+            List<GenericVisibleEvent> allFenceSC = program.getThreadEvents(GenericVisibleEvent.class).stream()
+                    .filter(e -> e.hasTag(Tag.FENCE) && e.hasTag(Tag.PTX.SC))
```

Comment:
Can we use `List<Event>` and go over all(!) events and filter for those that have `Visible`, `Fence` and `PTX.SC` tag?